### ISSUE TITLE
Application list doesn't load more results if the window is taller than the list

### ISSUE THREAD

**Issue by hewIngram** [pop-os/shop#233]
**Distribution (run `cat /etc/os-release`):**
 ```
NAME="Pop!_OS"
VERSION="20.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.10"
VERSION_ID="20.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=groovy
UBUNTU_CODENAME=groovy
LOGO=distributor-logo-pop-os
```

 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
pop-shop
`3.3.0pop0~1600965555~20.10~f37e26a`

**Issue/Bug Description:**
Opening any of the category app lists or when searching in the searchbox the app list view it loads 21 apps but on screens where the window is tall enough that the bottom of the list is inside  the height of the window the pagination doesn't work so you can't see more than those 21 results. 
A screenshot hopefully makes more sense: 

![pop_shop-list-length](https://user-images.githubusercontent.com/17004905/98452135-7cc96c00-2144-11eb-8643-5bd7f6cefdc6.png)

**Steps to reproduce (if you know):**
Open a pop-shop category in a tall window

**Expected behavior:**
It loads enough apps to fill the height of the window and pagination works as expected so all apps can be viewed

**Other Notes:**

**Comment by hewIngram** [comment]
I would be super happy to look at fixing this (will have a little look now) as it's probably a simple fix (or could be) and it'd be nice to make a first contribution to Pop as I've been enjoying using it so much!

My approach would probably be just to bump the default number of apps loaded which isn't really that elegant - if any of the maintainers have a view on that as a fix or a pointer to a nicer solution that'd be awesome.

**Comment by cassidyjames** [comment]
@hewIngram also note that Pop!_Shop is a fork of elementary AppCenter; this fix would probably be best made there which would then trickle back down to Pop!_OS: [@elementary/appcenter](https://github.com/elementary/appcenter)

**Comment by cassidyjames** [comment]
@davidmhewitt, I know you've spent a lot of time on AppCenter recently; any immediate ideas for fixing this?

**Comment by davidmhewitt** [comment]
If the page size on the scrolled vadjustment is equal to its upper bound, then we know all of the available rows are visible.

I've just tried to call the show more apps function when that's the case and it gets called correctly when more apps are needed and the rows are added, but for some reason they're not visible.

I don't really have much time to work on this during this week though. It would be good to get a duplicate issue opened upstream because I can confirm it.

**Comment by jacobgkau** [comment]
Upstream has closed their issue because they switched category views to use a grid, which isn't affected by the issue: https://github.com/elementary/appcenter/issues/1445 

![Screenshot from 2023-10-09 12-27-35](https://github.com/pop-os/shop/assets/7199422/834b6d84-453d-445e-9872-2cf28fc95652)

However, search results are still affected:

![Screenshot from 2023-10-09 12-27-57](https://github.com/pop-os/shop/assets/7199422/efc2b263-16ec-4a11-a72b-21bfa1daf63d)

**Comment by jacobgkau** [comment]
Upstream seems to have fixed the issue with https://github.com/elementary/appcenter/pull/2076, which was included in the 7.4.0 release, so we'll need to rebase on that version or higher to close this issue.

**Comment by jacobgkau** [comment]
This will be fixed by #437:

![Screenshot from 2024-03-05 18-08-53](https://github.com/pop-os/shop/assets/7199422/55897da7-0352-4bdd-aa96-01d83c748fff)
